core/src/main/scala/kafka/server/metadata/ClientQuotaCache.scala

```diff
@@ -122,6 +122,14 @@ class ClientQuotaCache {
       entityFilters.put(entityType, entityMatch)
     }
 
+    // Special case for non-strict empty filter, match everything
```

Comment:
What happens if we have an empty filter list and strict mode set?  It looks like an exception is thrown-- that doesn't seem correct?

Reply:
It should short-circuit and return an empty map on [L134](https://github.com/apache/kafka/pull/10254/files#diff-99bc678b86ad25d99b32bc192428120d2ff3f3d478159e2c353f4f5346b43718R133-R135)

Reply:
ok